test/box-luatest/gh_4348_transactional_ddl_test.lua

```diff
@@ -0,0 +1,281 @@
+            box.space.s:drop()
+        end
+        if box.space.s_renamed ~= nil then
+            box.space.s_renamed:drop()
+        end
+        if box.space.fs1 ~= nil then
+            box.space.fs1:drop()
+        end
+        if box.space.fs2 ~= nil then
+            box.space.fs2:drop()
+        end
+        if box.space.fs3 ~= nil then
+            box.space.fs3:drop()
+        end
+        if box.func.constr1 ~= nil then
+            box.func.constr1:drop()
+        end
+        if box.func.constr2 ~= nil then
+            box.func.constr2:drop()
+        end
+    end)
+end)
+
+g.test_transactional_ddl = function(cg)
+    cg.server:exec(function(engine)
+        ------------------------------------------------------------------------
+        -- box.schema.space.create ---------------------------------------------
+        ------------------------------------------------------------------------
+
+        local value
+        local s = box.schema.space.create('s', {engine = engine})
+        local fs1 = box.schema.space.create('fs1', {engine = engine})
+        local fs2 = box.schema.space.create('fs2', {engine = engine})
+        local fs3 = box.schema.space.create('fs3', {engine = engine})
+
+        s:create_index('spk', {parts = {{1, 'scalar'}}})
+        fs1:create_index('fs1pk', {parts = {{1, 'scalar'}}})
+        fs2:create_index('fs2pk', {parts = {{1, 'scalar'}}})
+        fs3:create_index('fs3pk', {parts = {{1, 'scalar'}}})
+
+        s:create_index('iut', {
+            parts = {{1, 'scalar'}},
+            unique = true,
+            type = 'tree'
+        })
+
+        s:create_index('int', {
+            parts = {{1, 'scalar'}},
+            unique = false,
+            type = 'tree'
+        })
+
+        if engine ~= 'vinyl' then
+            s:create_index('iuh', {
+                parts = {{1, 'scalar'}},
+                unique = true,
+                type = 'hash'
+            })
+        end
+
+        local fs1sk = fs1:create_index('fs1sk', {
+            parts = {{1, 'unsigned'}},
+            unique = false
+        })
+
+        box.schema.func.create('constr1', {
+            language = 'LUA',
+            is_deterministic = true,
+            body = 'function(t, c) return t.id < 999999 end'
+        })
+
+        box.schema.func.create('constr2', {
+            language = 'LUA',
+            is_deterministic = true,
+            body = 'function(t, c) return t.id < 99999 end'
+        })
+
+        -- Fill the spaces.
+        for i = 0, 1001  do -- More than 1k to ensure yielding.
+            fs1:insert({i + 1})
+            fs2:insert({i + 2})
+            fs3:insert({i + 3})
+            s:insert({i, i + 1, i + 2, i + 3})
+        end
+
+        ------------------------------------------------------------------------
+        -- box.schema.space.format ---------------------------------------------
+        ------------------------------------------------------------------------
```

Comment:
It'd be better to split the test into several test cases instead of using comments.

Reply:
Splitted.